test-galasactl-ecosystem.sh

```diff
@@ -189,6 +189,151 @@ function launch_test_on_ecosystem_with_portfolio {
     success "Submitting test to ecosystem worked OK"
 }
 
+#--------------------------------------------------------------------------
+function runs_download_check_folder_names_during_test_run {
+    h2 "Performing runs download..."
+
+    run_name=$1
+
+    mkdir -p ${BASEDIR}/temp
+    cd ${BASEDIR}/temp
+
+    # Create the portfolio.
+    cmd="${BASEDIR}/bin/${binary} runs prepare \
+    --bootstrap $bootstrap \
+    --stream inttests \
+    --portfolio portfolio.yaml \
+    --test ${GALASA_TEST_NAME_SHORT} \
+    --log -"
+
+    info "Command is: $cmd"
+
+    $cmd
+    rc=$?
+    # We expect a return code of '0' because this test is in the ecosystem's testcatalog.
+    if [[ "${rc}" != "0" ]]; then 
+        error "Failed to create a portfolio with a known test from the ecosystem's testcatalog."
+        exit 1
+    fi
+    success "Creating portfolio.yaml worked OK"
+
+    h2 "Launching test on an ecosystem from a portfolio..."
+
+    cd ${BASEDIR}/temp
+
+    log_file="runs-submit-output.txt"
+
+    cmd="${BASEDIR}/bin/${binary} runs submit \
+    --bootstrap ${bootstrap} \
+    --portfolio portfolio.yaml \
+    --throttle 1 \
+    --poll 10 \
+    --progress 1 \
+    --noexitcodeontestfailures \
+    --log ${log_file}"
+
+    set -o pipefail # Fail everything if anything in the pipeline fails. Else we are just checking the 'tee' return code.
+
+    
+    $cmd &
+
+    is_done="false"
+    retries=0
+    max=100
+    target_line=""
+    
+    
+    while [[ "${is_done}" == "false" ]]; do
+        if [[ -e $log_file ]]; then
+            echo "file exists"
+            target_line=$(cat ${log_file} | grep "submitted")
+            
+
+            if [[ "$target_line" != "" ]]; then
+                echo "Target line is found."
+                is_done="true"
+            fi
+        fi    
+        sleep 1
+        ((retries++))
+        if (( $retries > $max )); then 
+            echo "Too many retries."
+            is_done="true"
+        fi
+    done
+
+    run_name=$(echo $target_line | cut -f4 -d' ')
+    echo "Run name is $run_name"
+
+    rc=$?
```

Comment:
This is picking up the return code from the 'echo' command above.
Are you trying to get the rc of the process we put into the background ?